src/Form/Form.php

```diff
@@ -421,6 +406,45 @@ static function (mixed $value, Widget $widget, self $form) use ($fieldConfig, $f
         return $this;
     }
 
+    /**
+     * Create the data container mock.
+     */
+    private function createDataContainerMock(mixed $value, string $field, string $name): DataContainer
+    {
+        return new class($this->getBoundModel(), $value, $field, $name) extends DataContainer {
+            private ?Model $model;
+
+            public function __construct($model, $value, $field, $name)
+            {
+                $this->id = $model ? $model->id : 0;
+                $this->table = $model ? $model::getTable() : '';
+                $this->value = $value;
+                $this->field = $field;
+                $this->inputName = $name;
+                $this->activeRecord = $model; // BC
```

Comment:
`activeRecord` is not BC, it just has a more clear meaning in Contao 5: `getCurrentRecord()` should always return the record as it is currently stored in the DB while `activeRecord` contains the unsafed changes. 

So I think your logic currently represents the same values for both cases which would be wrong afaics.

Reply:
OK, then I'll wait with the testing til the feedback.

Reply:
Changed in 2f16ebfc. I added the BC comment because `activeRecord` is deprecated as of Contao 5.0.